Python/sysmodule.c

```diff
@@ -222,8 +222,8 @@ sys_audit_tstate(PyThreadState *ts, const char *event,
         PyDTrace_AUDIT(event, (void *)eventArgs);
     }
 
-    /* Call interpreter hooks */
-    if (is->audit_hooks) {
+    /* Call interpreter hooks if the GIL is held */
+    if (PyGILState_Check() && is->audit_hooks) {
```

Comment:
is this related to the issue? it looks like a bugfix. best mention that in the logs/news if there was a problem before.

Reply:
It's related because we (apparently) never triggered any hooks from outside the GIL before. Now we potentially do, because PyGILState_Ensure may allocate a new threadstate, and while it _theoretically_ has the GIL, it isn't ready to call into any code. 

But I'm pretty sure nobody else could ever get into this state apart from us.

Reply:
I think after my other changes we don't need this. Turns out, calling into any audit hooks at all with an invalid GIL is a bad idea 😅